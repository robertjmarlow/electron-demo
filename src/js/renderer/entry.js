const { ipcRenderer } = require('electron');
const $ = require('jquery');
const os = require('os');
require('../../css/main.css');

$(document).ready(function() {
  ipcRenderer.on('getFileContents', function(event, arg) {
    $('#file-contents').append('<div class="fileContents">' + arg.toString().replace(/\n/g, '<br/>').replace(/ /g, '&nbsp') + '</div>');
  });

  ipcRenderer.on('getDirectoryContents', function(event, arg) {
    // yes, this was broken in the last checkin...
    $('#dir-contents').append('<div class="fileList">' + arg + '</div>');
  });

  ipcRenderer.send('getFileContents', './main.js');
  ipcRenderer.send('getDirectoryContents', '.');

  $('#os-stuff')
    .append('<div>Your OS is: <span class="big">' + os.platform() + ' ' + os.release() + '</span></div>')
    .append('<div>Your uptime (in seconds) is: <span class="big">' + os.uptime() + '</span></div>')
});
