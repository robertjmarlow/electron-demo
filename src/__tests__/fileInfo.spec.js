import React from 'react';
import renderer from 'react-test-renderer';
import FileInfo from '../jsx/FileInfo';

jest.mock('electron');

describe('File Info', () => {
  it('shows file contents', () => {
    const tree = renderer.create(
      <FileInfo/>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
