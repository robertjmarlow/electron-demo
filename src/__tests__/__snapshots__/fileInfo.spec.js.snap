exports[`File Info shows file contents 1`] = `
<div
  className="fileContents">
  <div>
    this
  </div>
  <div>
     
  </div>
  <div>
     
  </div>
  <div>
     
  </div>
  <div>
     
  </div>
  <div>
     
  </div>
  <div>
    is a file
  </div>
  <div>
    with some stuff in it.
  </div>
</div>
`;
